atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueService.java

```diff
@@ -889,6 +887,29 @@ public String toString() {
         }
     }
 
+    private void truncateInternal(Client client, String tableName) throws TException {
+        for (int tries = 1; tries <= CassandraConstants.MAX_TRUNCATION_ATTEMPTS; tries++) {
+            boolean successful = true;
+            try {
+                client.truncate(internalTableName(tableName));
+            } catch (TException e) {
+                log.error("Cluster was unavailable while we attempted a truncate for table " + tableName + "; we will try " + (CassandraConstants.MAX_TRUNCATION_ATTEMPTS - tries) + " additional time(s). (" + e.getMessage() + ")");
+                if (CassandraConstants.MAX_TRUNCATION_ATTEMPTS - tries == 0) {
+                    throw e;
+                }
+                successful = false;
+                try {
+                    Thread.sleep(new Random().nextInt((1 << (CassandraConstants.MAX_TRUNCATION_ATTEMPTS - tries)) - 1) * 1000);
```

Comment:
Is this supposed to be a reverse exponential backoff and rounded to a second (pauses currently get shorter after more tries)? Probably worth extracting this out and not creating a new Random each try.